docs/src/design/lookups/main.md

```diff
@@ -0,0 +1,55 @@
+2. To prove correct interaction between two independent sections of the execution trace, e.g., between the main trace where the result of some operation is required, but would be expensive to compute, and a specialized component which can perform that operation cheaply.
+
+The first is achieved using [virtual tables](#virtual-tables-in-miden-vm) of data, where we add a row at some cycle in the trace and remove it at a later cycle when it is needed again. Instead of maintaining the entire table in the execution trace, multiset checks allow us to prove data consistency of this table using one running product column.
+
+The second is done by reducing each operation to a lookup value and then using a [communication bus](#communication-buses-in-miden-vm) to provably connect the two sections of the trace. These communication buses can be implemented either via [multiset checks](./multiset.md#communication-buses) or via the [LogUp argument](./logup.md).
+
+
+## Virtual tables in Miden VM
+
+Miden VM makes use of 6 virtual tables across 4 components, all of which are implemented via [multiset checks](./multiset.md#virtual-tables):
+
+- Stack:
+    - [Overflow table](../stack/main.md#overflow-table)
+- Decoder:
+    - [Block stack table](../decoder/main.md#block-stack-table)
+    - [Block hash table](../decoder/main.md#block-hash-table)
+    - [Op group table](../decoder/main.md#op-group-table)
+- Chiplets:
+    - [Chiplets virtual table](../chiplets/main.md#chiplets-virtual-table), which combines the following two tables into one:
+        - [Hash chiplet sibling table](../chiplets/hasher.md#sibling-table-constraints)
+        - [Kernel ROM chiplet procedure table](../chiplets/kernel_rom.md#kernel-procedure-table-constraints)
+
+## Communication buses in Miden VM
+
+One strategy for improving the efficiency of a zero knowledge virtual machine is to use specialized components for complex operations and have the main circuit “offload” those operations to the corresponding components by specifying inputs and outputs and allowing the proof of execution to be done by the dedicated component instead of by the main circuit.
+
+These specialized components are designed to prove the internal correctness of the execution of the operations they support. However, in isolation they cannot make any guarantees about the source of the input data or the destination of the output data.
+
+In order to prove that the inputs and outputs specified by the main circuit match the inputs and outputs provably executed in the specialized component, some kind of provable communication bus is needed.
+
+This bus is typically implemented as some kind of lookup argument, and in Miden VM in particular we use multiset checks or LogUp.
+
+Miden VM uses 2 communication buses:
+
+- The chiplets bus [$b_{chip}$](../chiplets/main.md#chiplets-bus), which communicates with all of the chiplets (Hash, Bitwise, and Memory). It is implemented using multiset checks.
```

Comment:
nit: we should probably mention Kernel ROM chiplet here too.

Reply:
Good point. I can update this and address the multiplicity comment together once I know your thought on that one. If you want to merge this first that's also fine & I can address both points in a small follow-up PR